### PR TITLE
Add ability to zoom images on web/desktop

### PR DESCRIPTION
**Pull request by marcaaron**
### Details
Adds ability to zoom to 200% by clicking on image attachment on web/mWeb

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/146569

### Tests
**web/desktop**
1. Click on an attachment
2. Verify there is a zoom cursor
3. Click and verify the image zooms
4. Hold and drag and verify the image pans
5. Click again and verify the image zooms back out

**mWeb**
1. Click on an attachment in mobile web browser
1. Verify that pinch to zoom works normally

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [ ] iOS
- [ ] Android

#### Web
https://user-images.githubusercontent.com/32969087/112687216-43577b80-8e1b-11eb-9963-e342373c804e.mp4

#### Desktop
https://user-images.githubusercontent.com/32969087/112689955-10af8200-8e1f-11eb-9033-a1c8c61f26ed.mp4

### CODE REVIEW

**Comment by marcaaron** [comment]
Tagging in @roryabraham too since he's got some experience with UI widget guys in E.cash 😄

**Comment by roryabraham** [comment]
@marcaaron I was testing this on mobile web on my iPhone and noticed it had some sort of unintended consequences. It's a bit hard to describe, so I suggest you try it by creating an ngrok tunnel to your localhost:8080 and visiting that tunnel on your phone.

On iPhone on mobile web and iOS right now, you can zoom into the images normally, without the changes in this PR. But with this PR, if you do that normal "pinch zoom", it recognizes it as a mouse-click, zooms the image (I think where you first touched the screen), and has a weird jumping effect.

I think basically we want to make sure we're not doing anything on tap, while on mouse-click we want to zoom in.

File: src/components/ImageView/index.js [hunk 1]
```diff
@@ -1,35 +1,117 @@
...
+            initialX: 0,
+            initialY: 0,
+        };
+    }
 
-ImageView.propTypes = propTypes;
-ImageView.displayName = 'ImageView';
+    componentDidMount() {
+        document.addEventListener('mousemove', this.trackMouseMove.bind(this));
+        document.addEventListener('touchmove', this.trackMouseMove.bind(this));
```
Comment (roryabraham) [inline src/components/ImageView/index.js:28]: Why did you need to set up a listener for `touchmove`?

**Comment by marcaaron** [comment]
Thanks for testing that. I guess my thinking here was that we sort of unofficially support pinching and zooming now, but that's probably more useful on mWeb than what these changes are providing to desktop/web users. So, maybe we should just disable this for mobile web and let people pinch/zoom normally for now? Not sure if there is an easy way to identify web vs. mobile web besides "screen width".

File: src/components/ImageView/index.js (hunk 1 above)
Comment (marcaaron) [reply src/components/ImageView/index.js:28]: So that mobile web touches would set `isDragging`

Comment (marcaaron) [reply src/components/ImageView/index.js:28]: But maybe there is a solution where we don't need to do this and we can tell what kind of mouse event we've got.

**Comment by shawnborton** [comment]
Nice!

**Comment by marcaaron** [comment]
@roryabraham came up with a temporary solution for the mobile web issue by adding a utility that detects if we have a touch screen. If we do, we'll assume that we can use that to zoom in on images and not utilize the feature, if not, then we'll allow the mouse zoom and drag feature instead. Let me know what you think.

**Comment by marcaaron** [comment]
I will also caveat this by saying I think the long term solution here will be to use a single component for native/mWeb eventually. But didn't want to dive down that rabbit hole at the expense of quickly improving the desktop/web experience.

File: src/components/ImageView/index.js [hunk 2]
```diff
@@ -1,35 +1,145 @@
...
 
-const ImageView = props => (
-    <View
-        style={[
+class ImageView extends PureComponent {
+    constructor(props) {
+        super(props);
+        this.scrollableRef = null;
+        this.canUseTouchScreen = canUseTouchScreen();
+        this.containerStyles = [
```
Comment (roryabraham) [inline src/components/ImageView/index.js:17]: Only NAB is it seems like this should be put in a style file instead of declared in the constructor.

**Review by roryabraham — APPROVED** [review]
> the long term solution here will be to use a single component for native/mWeb eventually

Interesting. Agreed we don't need to go down this rabbit hole today, but were you thinking another platform file extension for mWeb?

**Comment by marcaaron** [comment]
>but were you thinking another platform file extension for mWeb?

Ah no, I'm suggesting that native and mobile web use the same exact component to handle touches + pinch zoom with something like [this](https://docs.swmansion.com/react-native-gesture-handler/docs/api/gesture-handlers/pan-gh/) which has web support, but still has a few kinks to work out.

**Comment by roryabraham** [comment]
Oh, nice. Yeah that makes more sense to me. Anyways, this LGTM and tested well 👍
